zerver/views/streams.py

```diff
@@ -393,18 +395,21 @@ def update_stream_backend(
             check_stream_name_available(user_profile.realm, new_name)
         do_rename_stream(stream, new_name, user_profile)
 
+    nobody_group = get_system_user_group_by_name(SystemGroups.NOBODY, user_profile.realm_id)
     request_settings_dict = locals()
     for setting_name, permission_configuration in Stream.stream_permission_group_settings.items():
         assert setting_name in request_settings_dict
         if request_settings_dict[setting_name] is None:
             continue
 
         setting_value = request_settings_dict[setting_name]
-        new_setting_value = parse_group_setting_value(setting_value.new)
+        new_setting_value = parse_group_setting_value(setting_value.new, nobody_group)
 
         expected_current_setting_value = None
         if setting_value.old is not None:
-            expected_current_setting_value = parse_group_setting_value(setting_value.old)
+            expected_current_setting_value = parse_group_setting_value(
```

Comment:
Parsing the empty group value to nobody group ID here will result in another query to get the group in `access_user_group_for_setting` but this seems a cleaner way to do this. Webapp anyways make sure that we pass nobody group ID when the pill container is empty, so this case will be possible only when making requests other than webapp.

Reply:
Plus, `access_user_group_for_setting` handles locking of group objects.

Reply:
Yeah, I think a bit of extra work for a situation that shouldn't happen is totally fine. We only care about avoiding +1 queries in loops, for high-volume requests. and things with DoS risk, and this is none of those.

Reply:
Another way to do this would be to make `parse_group_setting_value` take a second `realm` parameter, and only look up the `role:nobody` group from the `Realm` in the case that we need it.

But that is awkward given the loop here, so we'd need a cache, and that's starting to get silly to avoid a cheap query in a rare operation.